File: Softing/Files/OPCToolbox/V43x/Windows/NET/samples/server/ServerDLL/CS/TestDll/Console.cs
//-----------------------------------------------------------------------------
//                                                                            |
//                   Softing Industrial Automation GmbH                       |
//                        Richard-Reitzner-Allee 6                            |
//                           85540 Haar, Germany                              |
//                                                                            |
//                 This is a part of the Softing OPC Toolbox                  |
//       Copyright (c) 1998 - 2012 Softing Industrial Automation GmbH         |
//                           All Rights Reserved                              |
//                                                                            |
//-----------------------------------------------------------------------------
//-----------------------------------------------------------------------------
//                             OPC Toolbox C#                                 |
//                                                                            |
//  Filename    : Console.cs                                                  |
//  Version     : 4.31                                                        |
//  Date        : 01-August-2012                                              |
//                                                                            |
//  Description : Console application main implementation                     |
//                                                                            |
//-----------------------------------------------------------------------------
using System;
using System.Collections;
using System.Runtime.InteropServices;
using System.Threading;
using System.IO;
using Softing.OPCToolbox;

//	Add the dll sample namespace
using Softing.OPCToolbox.DllSample;

namespace TestDll
{
	public class Console
	{
		
		public static void Main(String []args)
		{
			try{
				//create an instance of the generated DLL
				Dll usedDll = new Dll();
				
				// call the DLL's Start method; server will be initialized, 
				// namespaces will be created and simulation thread will be started
				int result  = usedDll.Start();

				if (result != (int)EnumResultCode.S_OK)
				{					
					return;
				}

				System.Console.WriteLine("Press \'e\' or \'q\' and then \'ENTER\'to exit");

				bool end = false;

				while (!end)
				{
					String read = System.Console.ReadLine();
					switch (read)
					{
						case "E":
						case "e":
						case "Q":
						case "q":
							end = true;							
							break;

						default: 						
							break;
					}					
				}// while

				//calls the DLL's Stop method which terminates the server application and closes the simulation thread
				usedDll.Stop();

			}
			catch(Exception exc)
			{
				System.Console.WriteLine(exc.ToString());
			}
		}	//	end Main
	
	}	//	end class Console
}	//	end namespace

File: Softing/Files/OPCToolbox/V43x/Windows/C++/samples/client/tutorial/DA/Step3/Console.cpp
//-----------------------------------------------------------------------------
//                                                                            |
//                   Softing Industrial Automation GmbH                       |
//                        Richard-Reitzner-Allee 6                            |
//                           85540 Haar, Germany                              |
//                                                                            |
//                 This is a part of the Softing OPC Toolbox                  |
//       Copyright (c) 1998 - 2012 Softing Industrial Automation GmbH         |
//                           All Rights Reserved                              |
//                                                                            |
//-----------------------------------------------------------------------------
//-----------------------------------------------------------------------------
//                             OPC Toolbox C++                                |
//                                                                            |
//  Filename    : Console.cpp                                                 |
//  Version     : 4.31                                                        |
//  Date        : 01-August-2012                                              |
//                                                                            |
//  Description : Console application main implementation                     |
//                                                                            |
//-----------------------------------------------------------------------------
#include "stdafx.h"


#include "OpcClient.h"


using namespace std;

HANDLE g_endEvent = 0;


BOOL WINAPI ControlHandler(IN unsigned long crtlType)   // control type
{
	switch (crtlType)
	{
	case CTRL_BREAK_EVENT:  // use Ctrl+C or Ctrl+Break to end server
	case CTRL_C_EVENT:
	case CTRL_CLOSE_EVENT:
		if (g_endEvent)
		{
			SetEvent(g_endEvent);
		}

		return TRUE;
	}

	return FALSE;
}   //  end controlHandler


int _tmain(int argc, _TCHAR* argv[])
{
	long result = S_OK;
	g_endEvent = CreateEvent(NULL, FALSE, FALSE, NULL);
	SetConsoleCtrlHandler(ControlHandler, TRUE);
	//  create the OpcClient instance
	createOpcClient();
	OpcClient* pClient = getOpcClient();
	CoInitializeEx(NULL, COINIT_MULTITHREADED);
	CoInitializeSecurity(0, -1, 0, 0, RPC_C_AUTHN_LEVEL_NONE,
						 RPC_C_IMP_LEVEL_IDENTIFY, 0, EOAC_NONE, 0);

	//  TODO - design time license activation
	//  Fill in your design time license activation keys here
	//
	//  NOTE: you can activate one or all of the features at the same time
	//  firstly activate the COM-DA Client feature
	//result = getApplication()->activate(EnumFeature_DA_CLIENT, _T("XXXX-XXXX-XXXX-XXXX-XXXX"));

	if (!SUCCEEDED(result))
	{
		return result;
	}

	//  activate the XML-DA Client Feature
	//result = getApplication()->activate(EnumFeature_XMLDA_CLIENT, _T("XXXX-XXXX-XXXX-XXXX-XXXX"));

	if (!SUCCEEDED(result))
	{
		return result;
	}

	//  END TODO - design time license activation

	//  initialize the client instance
	if (!SUCCEEDED(pClient->initialize()))
	{
		pClient->terminate();
		destroyOpcClient();
		::CoUninitialize();
		CloseHandle(g_endEvent);
		return 1;
	}

	//  initialize the DA client simulation
	result |= pClient->initializeDaObjects();

	//  start the simulation thread
	if (FAILED(result))
	{
		return 0;
	}   //  end if

	_tprintf(_T("Press Ctrl-C to exit\n"));
	unsigned long waitTime = 100;
	BOOL end = FALSE;
	unsigned long waitRet = 0;

	while (!end)
	{
		waitRet = WaitForSingleObject(g_endEvent, waitTime);
		waitTime = 1000;

		if (waitRet == WAIT_OBJECT_0)
		{
			end = TRUE;
		}
		else
		{
			//  TODO: place your cyclic code here
			tstring value = pClient->readItem();
			_tprintf(_T("%s"), value.c_str());
		}   //  end if...else
	}   //  ens while

	pClient->terminate();
	destroyOpcClient();
	CloseHandle(g_endEvent);
	CoUninitialize();
	return 0;
}   //  end main
